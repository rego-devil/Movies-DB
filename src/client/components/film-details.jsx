import React from 'react';
import { Link } from 'react-router-dom';
import Loading  from './loading';

export class FilmDetails extends React.Component {

  componentWillMount() {
    this.props.getFilm(this.props.match.params.id);
  }

  render() {
    const {currentFilmDetails, onClearCurrentFilm, isFetching} = this.props;    

    if(!currentFilmDetails) 
      return null

    return (
      <div className="filmDetails">
       
        <img className="filmDetails__img"  src={currentFilmDetails.poster_path} />
        <div className="filmDetails__overview">
          <div className="filmDetails__top">
            <div className="filmDetails__title">{currentFilmDetails.title}</div>
            <div className="filmDetails__rating">{currentFilmDetails.vote_average}</div>
            
              <Link to="/">
                <button className="filmDetails__backToSearch" onClick={() => onClearCurrentFilm(null)}>
                  Back to search
                  </button>
              </Link>
            
          </div>
          <div className="filmDetails__middle">
            <div className="filmDetails__date">{currentFilmDetails.release_date && currentFilmDetails.release_date.split('-')[0]}</div>
            <div className="filmDetails__duration">{currentFilmDetails.runtime ? currentFilmDetails.runtime + ` min.` : ''}</div>
          </div>
          <div className="filmDetails__description">{currentFilmDetails.overview}</div>
          <div className="filmDetails__genres">
          {
            currentFilmDetails.genres && currentFilmDetails.genres.map((item, i) => (i > 0 ? ', ':'') + item)
          }
          </div>
        </div>
      </div>
      
      
      // currentFilmDetails ? (
      //     result
      // ) : (isFetching ? 
      //   <React.Fragment>
      //     { console.log('----3----')}
      //     <Loading />
      //   </React.Fragment> : '')
        
    )
  }
  
}