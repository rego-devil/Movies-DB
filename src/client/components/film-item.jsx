import React from 'react';
import {Context} from './context';

export class FilmItem extends React.Component { 
	constructor() {
		super();

		this._onShowFilmDetails = this._onShowFilmDetails.bind(this);
	}

	_onShowFilmDetails() {
		this.props.onShowFilmDetails(true);
	}

	render() {
		const {title, genre, date, imgUrl} = this.props;
		return (
			<div className="filmItem" onClick={this._onShowFilmDetails} >

				<img src={imgUrl} className="filmItem__img" />
				<div className="filmItem__description">
						<div className="filmItem__title">{title}</div>
						<div className="filmItem__releaseYear">{date}</div>
				</div>
				<div className="filmItem__genre">{genre}</div>

			</div>
		)
	}
	
}