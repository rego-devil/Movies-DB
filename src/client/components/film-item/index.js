export { default } from './film-item';
