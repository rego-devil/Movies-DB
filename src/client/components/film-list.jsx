import React from 'react';
import {FilmItem, FilmSort} from './';
import filmListJson from '../data/films.json';
import {Context} from './context';

export class FilmList extends React.Component {
	constructor() {
		super();
	}

	render() {
		const {searching, onShowFilmDetails} = this.props;
		return (
			!searching ? (
				<div className="noFilms">No films found</div> 
			) : (
				<React.Fragment>
					<FilmSort />
					<ul className="filmList">
						{
							filmListJson.map((item) => 
								<li key={item.id}>
									<FilmItem {...item} onShowFilmDetails={onShowFilmDetails} />
								</li>
							)
						}
					</ul>
				</React.Fragment>
			)
		) 
	}
}


