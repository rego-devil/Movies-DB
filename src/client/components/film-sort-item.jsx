import React from 'react';


export const SortItem = ({sortBy, onSort, currentSortField, value, sortOrderLocal }) => (
	<span 
		className={`filmSort__item ${currentSortField === sortBy ? 'filmSort__item--active' : ''}`} 
		onClick={() => onSort({sortBy: currentSortField, sortOrder: sortOrderLocal}) } 
	>
		{value}
		{currentSortField === sortBy ? sortOrderLocal === 'asc' ? String.fromCharCode(8593) : String.fromCharCode(8595) : ''  }
	</span>
)