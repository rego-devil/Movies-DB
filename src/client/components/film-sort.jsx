import React from 'react';

export const FilmSort = () => (
  <div className="filmInfo">
		<div className="filmInfo__count">7 film found</div>
		<div className="filmInfo__sortBy">Sort by: 
			<span className="filmInfo__sortByActive">release date</span>
			<span>rating</span>
		</div>
	</div>
)