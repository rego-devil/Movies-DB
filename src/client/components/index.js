export { Header } from './header';
export { FilmItem } from './film-item';
export { FilmList } from './film-list';
export { FilmDetails } from './film-details';
export { FilmSort } from './film-sort';
export { SortItem } from './film-sort-item';
export { Search } from './search';
export { Footer } from './footer';
