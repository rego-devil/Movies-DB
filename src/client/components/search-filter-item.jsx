import React from 'react';

export const SearchFilterItem = ({name, onActive, isActive}) => (
	<span className={`search__filterItem ${isActive? 'search__filterItem--active':''}`}  onClick={onActive}>
		{name}
	</span>
)