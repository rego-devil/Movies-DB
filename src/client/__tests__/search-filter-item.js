import React from 'react';
import {shallow, mount, render} from 'enzyme';
import {SearchFilterItem} from '../components/search-filter-item';
// import toJson from 'enzyme-to-json';

describe('Search Filter item component', () => {

    test('renders without crashing', () => {
        const wrapper = shallow(<SearchFilterItem />);
        expect(wrapper).toMatchSnapshot();
    });

    test('render component with prop', () => {
        const wrapper = shallow(<SearchFilterItem isActive={true} />);
        expect(wrapper.find('.search__filterItem').hasClass('search__filterItem--active')).toEqual(true);
    });
});