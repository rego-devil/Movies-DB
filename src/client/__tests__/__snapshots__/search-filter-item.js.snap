// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Filter item component renders without crashing 1`] = `
<span
  className="search__filterItem "
/>
`;
